src/pages/VaultsPage/VaultTransactions.tsx

```diff
@@ -1,62 +1,285 @@
+import { BigNumber } from 'ethers';
+import { getCircle, getVaultAndTransactions } from 'lib/gql/queries';
+import { decodeCircleId } from 'lib/vaults';
+import { DateTime } from 'luxon';
 import { useQuery } from 'react-query';
 import { useParams } from 'react-router-dom';
 import { styled } from 'stitches.config';
 
 import { Link, Panel, Text } from 'ui';
 import { OrgLayout, SingleColumnLayout } from 'ui/layouts';
+import { getProviderForChain } from 'utils/provider';
+
+import { Awaited } from 'types/shim';
+
+type VaultAndTransactions = Awaited<ReturnType<typeof getVaultAndTransactions>>;
 
 export const VaultTransactions = () => {
   const { address } = useParams();
-  const { isLoading, isIdle, data } = useQuery('vault', () =>
-    client.query(
-      {
-        vaults: [
-          { where: { vault_address: { _eq: address } } },
-          {
-            ...allVaultFields,
-            protocol: {
-              name: true,
-            },
-          },
-        ],
-      },
-      { operationName: 'getVault' }
-    )
-  );
-
-  const vault = data?.vaults[0];
+  const [vaultTxList, setVaultTxList] = useState<any>([]);
+  const {
+    isLoading,
+    isIdle,
+    data: vault,
+  } = useQuery(['vault', address], async () => {
+    const result = await getVaultAndTransactions(address);
+    return result;
+  });
 
-  if (!vault && !isLoading && !isIdle) {
+  useEffect(() => {
+    if (vault) getOnchainVaultTransactions(vault).then(setVaultTxList);
+  }, [vault]);
+  if (!vault) {
     // TODO
-    return <SingleColumnLayout>404</SingleColumnLayout>;
+    if (!isLoading && !isIdle)
+      return <SingleColumnLayout>404</SingleColumnLayout>;
+    return <>Loading...</>;
   }
 
   return (
-    <OrgLayout name={data?.vaults[0].protocol.name}>
+    <OrgLayout name={vault.protocol.name}>
       <Panel>
         <Text h2 css={{ mb: '$md' }}>
           All Transactions for {vault?.symbol?.toUpperCase()} Vault
         </Text>
-        <TransactionTable
-          rows={[
-            ...dummyTableData,
-            ...dummyTableData.map(r => ({
-              ...r,
-              hash: r.hash.replace('a', 'b'),
-            })),
-            ...dummyTableData.map(r => ({
-              ...r,
-              hash: r.hash.replace('a', 'c'),
-            })),
-          ]}
-        />
+        <TransactionTable rows={vaultTxList} />
       </Panel>
     </OrgLayout>
   );
 };
 
+export async function getOnchainVaultTransactions(vault: VaultAndTransactions) {
+  const { chain_id } = vault;
+  const provider = getProviderForChain(chain_id);
+  const contracts = new ContractsReadonly(chain_id, provider);
+  const depositEvents = getDepositEvents(contracts, vault);
+  const withdrawEvents = getWithdrawEvents(contracts, vault);
+  const distributionEvents = getDistributionEvents(contracts, vault);
+  const eventResults = await Promise.all([
+    depositEvents,
+    withdrawEvents,
+    distributionEvents,
+  ]);
```

Comment:
```suggestion
  const eventResults = await Promise.all([
    getDepositEvents(contracts, vault),
    getWithdrawEvents(contracts, vault),
    getDistributionEvents(contracts, vault),
  ]);
```

i like this better because i was initially misled into thinking the calls were sync, not async; and the const names don't really add anything since they're only referenced shortly after and are very similar to the method names